lib/features/user/data/models/user_model.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:coffeecard/features/occupation/data/models/occupation_model.dart';
+import 'package:coffeecard/features/occupation/domain/entities/occupation.dart';
+import 'package:coffeecard/features/user/domain/entities/user.dart';
+import 'package:coffeecard/generated/api/coffeecard_api.swagger.dart';
+import 'package:coffeecard/generated/api/coffeecard_api_v2.swagger.dart';
+
+class UserModel extends User {
+  const UserModel({
+    required super.id,
+    required super.name,
+    required super.email,
+    required super.privacyActivated,
+    required super.occupation,
+    required super.rankMonth,
+    required super.rankSemester,
+    required super.rankTotal,
+  });
+
+  factory UserModel.fromResponse(UserResponse response) {
+    final programmeDto =
+        ProgrammeResponse.fromJson(response.programme as Map<String, dynamic>);
```

Comment:
Same thing goes for this, in theory it should be an `OccupationResponse` and `occupationDto`